### ISSUE TITLE
System.DllNotFoundException: lib/linux/x86_64/libgit2-baa87df.so on Debian9

### ISSUE THREAD

**Issue by zmechu** [Splamy/TS3AudioBot#320]
Hello!

On Debian8 compilation works fine. On three instalation of Debian 9 still the same: "**System.DllNotFoundException: lib/linux/x86_64/libgit2-baa87df.so**"
where its exists:
```
find . -name libgit2-baa87df.so
./packages/gitversiontask/3.6.5/lib/linux/x86_64/libgit2-baa87df.so
./packages/GitVersionTask.3.6.5/lib/linux/x86_64/libgit2-baa87df.so
```
```
cat /etc/debian_version
9.4
```

```
# msbuild /p:Configuration=Release TS3AudioBot.sln
Microsoft (R) Build Engine version 15.6.0.0 ( Thu May 10 14:00:26 UTC 2018) for Mono
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 07/07/2018 21:42:06.
Project "/tmp/TS3AudioBot/TS3AudioBot.sln" on node 1 (default targets).
ValidateSolutionConfiguration:
  Building solution configuration "Release|Any CPU".
Project "/tmp/TS3AudioBot/TS3AudioBot.sln" (1) is building "/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj" (2) on node 1 (default targets).
Project "/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj" (2) is building "/tmp/TS3AudioBot/TS3Client/TS3Client.csproj" (3:2) on node 1 (default targets).
GenerateTargetFrameworkMonikerAttribute:
Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
CoreCompile:
Skipping target "CoreCompile" because all output files are up-to-date with respect to the input files.
_CopyFilesMarkedCopyLocal:
  Touching "/tmp/TS3AudioBot/TS3Client/obj/Release/TS3Client.csproj.CopyComplete".
CopyFilesToOutputDirectory:
  TS3Client -> /tmp/TS3AudioBot/TS3Client/bin/Release/TS3Client.dll
Done Building Project "/tmp/TS3AudioBot/TS3Client/TS3Client.csproj" (default targets).
GenerateTargetFrameworkMonikerAttribute:
Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
MSBUILD : warning : WARN [07/07/18 21:42:10:57] Could not determine assembly version: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.DllNotFoundException: lib/linux/x86_64/libgit2-baa87df.so [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at (wrapper managed-to-native) LibGit2Sharp.Core.NativeMethods.git_libgit2_init() [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.NativeMethods+LibraryLifetimeObject..ctor () [0x00006] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.NativeMethods..cctor () [0x00058] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :    --- End of inner exception stack trace --- [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Proxy.git_buf_free (LibGit2Sharp.Core.Handles.GitBuf buf) [0x00000] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Handles.GitBuf.Dispose () [0x00000] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Proxy.ConvertPath (System.Func`2[T,TResult] pathRetriever) [0x0002e] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Proxy.git_repository_discover (LibGit2Sharp.Core.FilePath start_path) [0x0000d] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Repository.Discover (System.String startingPath) [0x00006] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at GitVersion.GitPreparer.GetDotGitDirectory () [0x0000f] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at GitVersion.ExecuteCore.ExecuteGitVersion (System.String targetUrl, System.String dynamicRepositoryLocation, GitVersion.Authentication authentication, System.String targetBranch, System.Boolean noFetch, System.String workingDirectory, System.String commitId, GitVersion.Config overrideConfig) [0x00046] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at GitVersion.ExecuteCore.TryGetVersion (System.String directory, GitVersion.VersionVariables& versionVariables, System.Boolean noFetch, GitVersion.Authentication authentication) [0x00000] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning : WARN [07/07/18 21:42:10:58] Could not determine assembly version: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.DllNotFoundException: lib/linux/x86_64/libgit2-baa87df.so [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at (wrapper managed-to-native) LibGit2Sharp.Core.NativeMethods.git_libgit2_init() [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.NativeMethods+LibraryLifetimeObject..ctor () [0x00006] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.NativeMethods..cctor () [0x00058] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :    --- End of inner exception stack trace --- [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Proxy.git_buf_free (LibGit2Sharp.Core.Handles.GitBuf buf) [0x00000] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Handles.GitBuf.Dispose () [0x00000] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Proxy.ConvertPath (System.Func`2[T,TResult] pathRetriever) [0x0002e] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Proxy.git_repository_discover (LibGit2Sharp.Core.FilePath start_path) [0x0000d] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Repository.Discover (System.String startingPath) [0x00006] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at GitVersion.GitPreparer.GetDotGitDirectory () [0x0000f] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at GitVersion.ExecuteCore.ExecuteGitVersion (System.String targetUrl, System.String dynamicRepositoryLocation, GitVersion.Authentication authentication, System.String targetBranch, System.Boolean noFetch, System.String workingDirectory, System.String commitId, GitVersion.Config overrideConfig) [0x00046] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at GitVersion.ExecuteCore.TryGetVersion (System.String directory, GitVersion.VersionVariables& versionVariables, System.Boolean noFetch, GitVersion.Authentication authentication) [0x00000] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning : WARN [07/07/18 21:42:10:59] Could not determine assembly version: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.DllNotFoundException: lib/linux/x86_64/libgit2-baa87df.so [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at (wrapper managed-to-native) LibGit2Sharp.Core.NativeMethods.git_libgit2_init() [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.NativeMethods+LibraryLifetimeObject..ctor () [0x00006] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.NativeMethods..cctor () [0x00058] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :    --- End of inner exception stack trace --- [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Proxy.git_buf_free (LibGit2Sharp.Core.Handles.GitBuf buf) [0x00000] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Handles.GitBuf.Dispose () [0x00000] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Proxy.ConvertPath (System.Func`2[T,TResult] pathRetriever) [0x0002e] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Proxy.git_repository_discover (LibGit2Sharp.Core.FilePath start_path) [0x0000d] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Repository.Discover (System.String startingPath) [0x00006] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at GitVersion.GitPreparer.GetDotGitDirectory () [0x0000f] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at GitVersion.ExecuteCore.ExecuteGitVersion (System.String targetUrl, System.String dynamicRepositoryLocation, GitVersion.Authentication authentication, System.String targetBranch, System.Boolean noFetch, System.String workingDirectory, System.String commitId, GitVersion.Config overrideConfig) [0x00046] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at GitVersion.ExecuteCore.TryGetVersion (System.String directory, GitVersion.VersionVariables& versionVariables, System.Boolean noFetch, GitVersion.Authentication authentication) [0x00000] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
CoreCompile:
Skipping target "CoreCompile" because all output files are up-to-date with respect to the input files.
FodyTarget:
Skipping target "FodyTarget" because all output files are up-to-date with respect to the input files.
_CopyFilesMarkedCopyLocal:
  Touching "/tmp/TS3AudioBot/TS3AudioBot/obj/Release/TS3AudioBot.csproj.CopyComplete".
_CopyOutOfDateSourceItemsToOutputDirectory:
Skipping target "_CopyOutOfDateSourceItemsToOutputDirectory" because all output files are up-to-date with respect to the input files.
_CopyAppConfigFile:
  Copying file from "obj/Release/TS3AudioBot.exe.config" to "bin/Release/TS3AudioBot.exe.config".
CopyFilesToOutputDirectory:
  TS3AudioBot -> /tmp/TS3AudioBot/TS3AudioBot/bin/Release/TS3AudioBot.exe
AfterBuild:
  Copying file from "lib/x86/libopus.dll" to "bin/Release/x86/libopus.dll".
  Copying file from "lib/x64/libopus.dll" to "bin/Release/x64/libopus.dll".
FodyVerifyTarget:
    Fody: Found path to weavers file '/tmp/TS3AudioBot/TS3AudioBot/FodyWeavers.xml'.
    Fody:   Skipped Verifying assembly since it is disabled in configuration
    Fody:   Finished verification in 4ms.
Done Building Project "/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj" (default targets).
Project "/tmp/TS3AudioBot/TS3AudioBot.sln" (1) is building "/tmp/TS3AudioBot/TS3ABotUnitTests/TS3ABotUnitTests.csproj" (4) on node 1 (default targets).
GenerateTargetFrameworkMonikerAttribute:
Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
CoreCompile:
Skipping target "CoreCompile" because all output files are up-to-date with respect to the input files.
_CopyFilesMarkedCopyLocal:
  Copying file from "/tmp/TS3AudioBot/TS3AudioBot/bin/Release/TS3AudioBot.exe.config" to "bin/Release/TS3AudioBot.exe.config".
  Touching "/tmp/TS3AudioBot/TS3ABotUnitTests/obj/Release/TS3ABotUnitTests.csproj.CopyComplete".
_CopyOutOfDateSourceItemsToOutputDirectory:
Skipping target "_CopyOutOfDateSourceItemsToOutputDirectory" because all output files are up-to-date with respect to the input files.
CopyFilesToOutputDirectory:
  TS3ABotUnitTests -> /tmp/TS3AudioBot/TS3ABotUnitTests/bin/Release/TS3ABotUnitTests.dll
Done Building Project "/tmp/TS3AudioBot/TS3ABotUnitTests/TS3ABotUnitTests.csproj" (default targets).
Project "/tmp/TS3AudioBot/TS3AudioBot.sln" (1) is building "/tmp/TS3AudioBot/Ts3ClientTests/Ts3ClientTests.csproj" (5) on node 1 (default targets).
GenerateTargetFrameworkMonikerAttribute:
Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
CoreCompile:
Skipping target "CoreCompile" because all output files are up-to-date with respect to the input files.
_CopyFilesMarkedCopyLocal:
  Touching "/tmp/TS3AudioBot/Ts3ClientTests/obj/Release/Ts3ClientTests.csproj.CopyComplete".
_CopyAppConfigFile:
  Copying file from "obj/Release/Ts3ClientTests.exe.config" to "bin/Release/Ts3ClientTests.exe.config".
CopyFilesToOutputDirectory:
  Ts3ClientTests -> /tmp/TS3AudioBot/Ts3ClientTests/bin/Release/Ts3ClientTests.exe
Done Building Project "/tmp/TS3AudioBot/Ts3ClientTests/Ts3ClientTests.csproj" (default targets).
Done Building Project "/tmp/TS3AudioBot/TS3AudioBot.sln" (default targets).

Build succeeded.

"/tmp/TS3AudioBot/TS3AudioBot.sln" (default target) (1) ->
"/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj" (default target) (2) ->
(WriteVersionInfoToBuildLog target) ->
  MSBUILD : warning : WARN [07/07/18 21:42:10:57] Could not determine assembly version: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.DllNotFoundException: lib/linux/x86_64/libgit2-baa87df.so [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at (wrapper managed-to-native) LibGit2Sharp.Core.NativeMethods.git_libgit2_init() [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.NativeMethods+LibraryLifetimeObject..ctor () [0x00006] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.NativeMethods..cctor () [0x00058] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :    --- End of inner exception stack trace --- [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Proxy.git_buf_free (LibGit2Sharp.Core.Handles.GitBuf buf) [0x00000] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Handles.GitBuf.Dispose () [0x00000] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Proxy.ConvertPath (System.Func`2[T,TResult] pathRetriever) [0x0002e] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Proxy.git_repository_discover (LibGit2Sharp.Core.FilePath start_path) [0x0000d] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Repository.Discover (System.String startingPath) [0x00006] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at GitVersion.GitPreparer.GetDotGitDirectory () [0x0000f] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at GitVersion.ExecuteCore.ExecuteGitVersion (System.String targetUrl, System.String dynamicRepositoryLocation, GitVersion.Authentication authentication, System.String targetBranch, System.Boolean noFetch, System.String workingDirectory, System.String commitId, GitVersion.Config overrideConfig) [0x00046] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at GitVersion.ExecuteCore.TryGetVersion (System.String directory, GitVersion.VersionVariables& versionVariables, System.Boolean noFetch, GitVersion.Authentication authentication) [0x00000] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]


"/tmp/TS3AudioBot/TS3AudioBot.sln" (default target) (1) ->
"/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj" (default target) (2) ->
(UpdateAssemblyInfo target) ->
  MSBUILD : warning : WARN [07/07/18 21:42:10:58] Could not determine assembly version: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.DllNotFoundException: lib/linux/x86_64/libgit2-baa87df.so [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at (wrapper managed-to-native) LibGit2Sharp.Core.NativeMethods.git_libgit2_init() [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.NativeMethods+LibraryLifetimeObject..ctor () [0x00006] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.NativeMethods..cctor () [0x00058] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :    --- End of inner exception stack trace --- [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Proxy.git_buf_free (LibGit2Sharp.Core.Handles.GitBuf buf) [0x00000] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Handles.GitBuf.Dispose () [0x00000] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Proxy.ConvertPath (System.Func`2[T,TResult] pathRetriever) [0x0002e] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Proxy.git_repository_discover (LibGit2Sharp.Core.FilePath start_path) [0x0000d] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Repository.Discover (System.String startingPath) [0x00006] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at GitVersion.GitPreparer.GetDotGitDirectory () [0x0000f] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at GitVersion.ExecuteCore.ExecuteGitVersion (System.String targetUrl, System.String dynamicRepositoryLocation, GitVersion.Authentication authentication, System.String targetBranch, System.Boolean noFetch, System.String workingDirectory, System.String commitId, GitVersion.Config overrideConfig) [0x00046] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at GitVersion.ExecuteCore.TryGetVersion (System.String directory, GitVersion.VersionVariables& versionVariables, System.Boolean noFetch, GitVersion.Authentication authentication) [0x00000] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]


"/tmp/TS3AudioBot/TS3AudioBot.sln" (default target) (1) ->
"/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj" (default target) (2) ->
(GetVersion target) ->
  MSBUILD : warning : WARN [07/07/18 21:42:10:59] Could not determine assembly version: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.DllNotFoundException: lib/linux/x86_64/libgit2-baa87df.so [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at (wrapper managed-to-native) LibGit2Sharp.Core.NativeMethods.git_libgit2_init() [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.NativeMethods+LibraryLifetimeObject..ctor () [0x00006] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.NativeMethods..cctor () [0x00058] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :    --- End of inner exception stack trace --- [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Proxy.git_buf_free (LibGit2Sharp.Core.Handles.GitBuf buf) [0x00000] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Handles.GitBuf.Dispose () [0x00000] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Proxy.ConvertPath (System.Func`2[T,TResult] pathRetriever) [0x0002e] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Core.Proxy.git_repository_discover (LibGit2Sharp.Core.FilePath start_path) [0x0000d] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at LibGit2Sharp.Repository.Discover (System.String startingPath) [0x00006] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at GitVersion.GitPreparer.GetDotGitDirectory () [0x0000f] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at GitVersion.ExecuteCore.ExecuteGitVersion (System.String targetUrl, System.String dynamicRepositoryLocation, GitVersion.Authentication authentication, System.String targetBranch, System.Boolean noFetch, System.String workingDirectory, System.String commitId, GitVersion.Config overrideConfig) [0x00046] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]
MSBUILD : warning :   at GitVersion.ExecuteCore.TryGetVersion (System.String directory, GitVersion.VersionVariables& versionVariables, System.Boolean noFetch, GitVersion.Authentication authentication) [0x00000] in <ecba14039ea0427fac6be14e5af4f4d5>:0  [/tmp/TS3AudioBot/TS3AudioBot/TS3AudioBot.csproj]

    3 Warning(s)
    0 Error(s)

Time Elapsed 00:00:04.27
[ERROR] FATAL UNHANDLED EXCEPTION: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.DllNotFoundException: lib/linux/x86_64/libgit2-baa87df.so
  at (wrapper managed-to-native) LibGit2Sharp.Core.NativeMethods.git_libgit2_init()
  at LibGit2Sharp.Core.NativeMethods+LibraryLifetimeObject..ctor () [0x00006] in <ecba14039ea0427fac6be14e5af4f4d5>:0
  at LibGit2Sharp.Core.NativeMethods..cctor () [0x00058] in <ecba14039ea0427fac6be14e5af4f4d5>:0
   --- End of inner exception stack trace ---
  at LibGit2Sharp.Core.NativeMethods+LibraryLifetimeObject.Finalize () [0x00000] in <ecba14039ea0427fac6be14e5af4f4d5>:0
```

```
Mono JIT compiler version 5.12.0.226 (tarball Thu May  3 09:49:46 UTC 2018)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:           __thread
        SIGSEGV:       altstack
        Notifications: epoll
        Architecture:  amd64
        Disabled:      none
        Misc:          softdebug
        Interpreter:   yes
        LLVM:          supported, not enabled.
        GC:            sgen (concurrent by default)
```
```
apt list --installed | grep libgit

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

libgit2-24/stable,now 0.25.1+really0.24.6-1 amd64 [installed]
libgit2-dev/stable,now 0.25.1+really0.24.6-1 amd64 [installed]
```

```
apt list --installed | grep sharp

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

libmono-csharp4.0c-cil/stable-stretch,now 5.12.0.226-0xamarin3+debian9b1 all [installed,automatic]
libmono-microsoft-csharp4.0-cil/stable-stretch,now 5.12.0.226-0xamarin3+debian9b1 all [installed,automatic]
libmono-sharpzip4.84-cil/stable-stretch,now 5.12.0.226-0xamarin3+debian9b1 all [installed,automatic]
mono-csharp-shell/stable-stretch,now 5.12.0.226-0xamarin3+debian9b1 all [installed,automatic]
```
```
apt list --installed | grep nuget

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

nuget/stable-stretch,now 4.3.0.4406.bin-0xamarin3+debian9b1 all [installed]
```

Thank you for support!

**Comment by Flakebi** [comment]
Same as #234.
This is a harmless error, the build should finish successfully, only the `!version` command will not return a version.

On arch linux this error also happens, seems like it is the same on debian 9.
The reason (at least on arch) is that openssl is built without some (deprecated) features so the shipped `libgit2-baa87df.so`, which you found, fails to link (yeah, this results in a `file not found` error). Replacing it with the system libgit2 should work.

**Comment by zmechu** [comment]
Right :-) Thank you for info.
